sdks/java/harness/src/main/java/org/apache/beam/fn/harness/control/ProcessWideInstructionHandler.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.fn.harness.control;
+
+import org.apache.beam.model.fnexecution.v1.BeamFnApi;
+import org.apache.beam.model.pipeline.v1.MetricsApi;
+import org.apache.beam.sdk.metrics.MetricsContainer;
+import org.apache.beam.sdk.metrics.MetricsEnvironment;
+
+/**
+ * Processes {@link BeamFnApi.InstructionRequest}'s and {@link
+ * BeamFnApi.MonitoringInfosMetadataRequest} and {@link BeamFnApi.HarnessMonitoringInfosResponse}
+ *
+ * <p>These instructions are not associated with the currently processed bundles. They return
+ * MonitoringInfos payloads and MonitoringInfo metadata for "process-wide" metrics, which return
+ * metric values calculated over the life of the process.
+ */
+public class ProcessWideInstructionHandler {
+  public ProcessWideInstructionHandler() {}
+
+  public static BeamFnApi.InstructionResponse.Builder monitoringInfoMetadata(
+      BeamFnApi.InstructionRequest request) {
+    BeamFnApi.MonitoringInfosMetadataResponse.Builder response =
+        BeamFnApi.MonitoringInfosMetadataResponse.newBuilder();
+    response.putAllMonitoringInfo(
+        MonitoringInfoShortIdCache.getShortIdCache()
+            .getInfos(request.getMonitoringInfos().getMonitoringInfoIdList()));
```

Comment:
It seems that the only moment when a MInfo gets added to the ShortIdCache is when we call `getShortId` on it, right? I don't see `getShortId` getting called for the `monitoringInfoMetadata`, only for the `harnessMonitoringInfos` - so I wonder how do we get the monitoring infos from the other containers in this call?